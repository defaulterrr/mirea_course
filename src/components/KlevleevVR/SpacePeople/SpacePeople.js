import React, { useEffect, useState } from 'react';

const HYPER_LINK_STYLE = {
  textDecoration: 'none',
  color: 'white',
};

const getUpdateUrl = () => 'http://api.open-notify.org/astros.json';
const fetchData = async (dataUpdate) => {
  try {
    const response = await fetch(getUpdateUrl());
    const json = await response.json();
    dataUpdate({
      number: json.number,
      people: json.people,
      isLoaded: true,
    });
  } catch (error) {
    dataUpdate({
      number: 'Fetching has failed',
      isLoaded: false,
    });
  }
};

const SpacePeople = () => {
  const [data, setData] = useState({
    number: 'Fetching...',
    people: ['name', 'name', 'name', '', '', ''],
    isLoaded: false,
  });
  useEffect(() => {
    fetchData(setData);
  }, []);
  if (data.isLoaded) {
    return (
      <div>
        <a style={HYPER_LINK_STYLE} href="https://www.howmanypeopleareinspacerightnow.com/">How many people are in space right now?</a>
        <h1>{data.number}</h1>
        {data.people.map((chunk) => <div>{chunk.name}</div>)}
      </div>
    );
  }

  return (
    <div>{data.number}</div>
  );
};

export default SpacePeople;
