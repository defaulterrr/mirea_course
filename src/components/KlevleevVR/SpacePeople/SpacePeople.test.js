import React from 'react';
import { unmountComponentAtNode } from 'react-dom';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import SpacePeople from './SpacePeople';


let container = null;
beforeEach(() => {
  // подготавливаем DOM-элемент, куда будем рендерить
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  // подчищаем после завершения
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it('Loading indicator', () => {
  const { getByText } = render(<SpacePeople />);
  const linkElement = getByText('Fetching...');
  expect(linkElement).toBeInTheDocument();
});

test('SpacePeople Snapshot', () => {
  const direction = renderer.create(<SpacePeople />);
  expect(direction.toJSON()).toMatchSnapshot();
});
