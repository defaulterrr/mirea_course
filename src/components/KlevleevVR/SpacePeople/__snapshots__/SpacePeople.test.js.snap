// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpacePeople Snapshot 1`] = `
<div>
  Fetching...
</div>
`;
